pennylane/devices/default_qubit.py

```diff
@@ -171,47 +171,47 @@ def apply(self, operations, rotations=None, **kwargs):
                     "on a {} device.".format(operation.name, self.short_name)
                 )
 
-            self._apply_operation(operation)
+            if isinstance(operation, QubitStateVector):
+                self._apply_state_vector(operation.parameters[0], operation.wires)
+            elif isinstance(operation, BasisState):
+                self._apply_basis_state(operation.parameters[0], operation.wires)
```

Comment:
These didn't really fit into the idea of applying an operation to a state, since these _provide_ the initial state.

Reply:
Makes sense. One thing to consider would be to change them such that the state is returned rather than internally updated. This would be useful because then there would be no difference between applying a state preparation and applying an operation in terms of syntax.
```python
self._state = self._apply_state_vector(operation.parameters[0], operation.wires)
```

Reply:
We could name the functions  `self._create_basis_state(state, wires)` and `self._create_state_vector` and have them both return a state variable.

They aren't really "applying" something since it overrides what existed before, if anything existed before.

Reply:
Thanks! I agree with both suggestions, but suggest we defer them until after release as part of another look at `default.qubit`.